internal/provider/data_network.go

```diff
@@ -0,0 +1,62 @@
+package provider
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataNetwork() *schema.Resource {
+	return &schema.Resource{
+		Description: "`unifi_network` data source can be used to retrieve the ID for a network by name.",
+
+		ReadContext: dataNetworkRead,
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Description: "The ID of this AP group.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"site": {
+				Description: "The name of the site the user group is associated with.",
+				Type:        schema.TypeString,
+				Computed:    true,
+				Optional:    true,
+			},
+			"name": {
+				Description: "The name of the user group to look up.",
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "Default",
```

Comment:
Is this the name of the default network? It has been a while since I set this up. Not sure if this is another copy paste issue.